analytic_engine/src/table/version.rs

```diff
@@ -568,29 +579,35 @@ impl TableVersion {
             .total_memory_usage()
     }
 
-    /// Switch all mutable memtables or just return the suggested segment
+    /// Return the suggested segment
     /// duration if sampling memtable is still active.
     ///
     /// Returns a duration if a sampled segment duration needs to be persisted.
     ///
     /// REQUIRE: Do in write worker
-    pub fn switch_memtables_or_suggest_duration(&self) -> Option<Duration> {
-        self.inner
-            .write()
-            .unwrap()
-            .memtable_view
-            .switch_memtables_or_suggest_duration()
+    pub fn suggest_duration(&self) -> Option<Duration> {
+        self.inner.write().unwrap().memtable_view.suggest_duration()
+    }
+
+    /// Switch all mutable memtables
+    ///
+    /// Returns the maxium `SequenceNumber` in the mutable memtables needs to be
+    /// freezed.
+    ///
+    /// REQUIRE: Do in write worker
+    pub fn switch_memtables(&self) -> Option<SequenceNumber> {
+        self.inner.write().unwrap().memtable_view.switch_memtables()
     }
 
     /// Stop timestamp sampling and freezed the sampling memtable.
     ///
     /// REQUIRE: Do in write worker
```

Comment:
Remove such comments `/// REQUIRE: Do in write worker` in the memtable, because it can be modified concurrently.

Reply:
ok